cmd/zc_traverser_blob.go

```diff
@@ -71,14 +71,28 @@ type blobTraverser struct {
 	isDFS bool
 }
 
-func (t *blobTraverser) IsDirectory(isSource bool) (bool, error) {
+var NonErrorDirectoryStubOverlappable = errors.New("The directory stub exists, and can overlap.")
+
+func (t *blobTraverser) IsDirectory(isSource bool) (isDirectory bool, err error) {
 	isDirDirect := copyHandlerUtil{}.urlIsContainerOrVirtualDirectory(t.rawURL)
 
+	blobURLParts, err := blob.ParseURL(t.rawURL)
+	if err != nil {
+		return false, err
+	}
+
 	// Skip the single blob check if we're checking a destination.
 	// This is an individual exception for blob because blob supports virtual directories and blobs sharing the same name.
 	// On HNS accounts, we would still perform this test. The user may have provided directory name without path-separator
 	if isDirDirect { // a container or a path ending in '/' is always directory
-		return true, nil
+		if blobURLParts.ContainerName != "" && blobURLParts.BlobName == "" {
+			// If it's a container, let's ensure that container exists. Listing is a safe assumption to be valid, because how else would we enumerate?
```

Comment:
Might be good to also note that GetProperties is not being called because there is a possibility of not having those specific permissions.

Reply:
Hmmmm. Permissions are a good thought there, but that wasn't actually a part of the original thought.